ray-operator/controllers/ray/suite_helpers_test.go

```diff
@@ -33,7 +33,7 @@ func listResourceFunc(ctx context.Context, workerPods *corev1.PodList, opt ...cl
 
 		count := 0
 		for _, aPod := range workerPods.Items {
-			if (reflect.DeepEqual(aPod.Status.Phase, corev1.PodRunning) || reflect.DeepEqual(aPod.Status.Phase, corev1.PodPending)) && aPod.DeletionTimestamp == nil {
+			if (reflect.DeepEqual(aPod.Status.Phase, corev1.PodRunning) || reflect.DeepEqual(aPod.Status.Phase, corev1.PodPending)) && (aPod.DeletionTimestamp == nil || len(aPod.Finalizers) != 0) {
```

Comment:
Without this, the `listResourceFunc` will always return 0 even if Pods are not yet deleted because of their Finalizers, and make the following assertion fail.
```go
Eventually(listResourceFunc(ctx, &allPods, allFilters...), time.Second*3, time.Millisecond*500).
	Should(Equal(numPods), fmt.Sprintf("all pods %v", allPods.Items))
```

I found this failure very misleading and, most importantly, without the change, we can't make sure those Pods still exist.

Reply:
I don't think the `k8sClient.List` does anything special. The behavior you observed was expected but since the deletionTimestamp was not zero, the pod would not be counted to the return value of `listResourceFunc`.

That made `listResourceFunc` always return 0 even though those Pods did still exist. Therefore, I want it to count Pods with non-zero deletionTimestamps and finalizers, otherwise, the 0 returned is too misleading.